プロトタイプ投稿

hmiyazaki opened the pull request:
# WHY

プロトタイプの投稿がこのアプリの肝だから
# WHAT

プロトタイプを投稿できるようにした。
ただし…
- 投稿時に複数の画像（枚数が不定）を添付できない
- トップページに画像が表示されない

Commit:
comment, like, tagというモデルとそのテーブルを作成した

Commit:
既存のテーブルに変更を加えた
prototypes title カラムを追加
prototype_images カラム名を変更。prototype_images -> prototype_image

Commit:
プロトタイプが投稿できるようにした。
ただ、トップページにプロトタイプの画像が表示されません…

kuboshizuma commented on app/models/comment.rb:
```diff
@@ -0,0 +1,3 @@
+class Comment < ActiveRecord::Base
```
今回のプルリクに関係ないモデルなのでこのプルリクに含めるのではなくコメント機能を実装する際のプルリクに含めるのが良いですね。

hmiyazaki replied:
comment モデルは消しました。

kuboshizuma commented on app/models/comment.rb:
```diff
@@ -0,0 +1,3 @@
+class Comment < ActiveRecord::Base
+  belngs_to :prototype
```
ここスペルミスですね。
belngs_to => belongs_to

hmiyazaki replied:
ご指摘ありがとうございます。

kuboshizuma commented on app/models/like.rb:
```diff
@@ -0,0 +1,2 @@
+class Like < ActiveRecord::Base
```
このモデルもこのプルリクには必要なさそうです。

hmiyazaki replied:
likeモデルも消しました

kuboshizuma commented on app/models/tag.rb:
```diff
@@ -0,0 +1,3 @@
+class Tag < ActiveRecord::Base
```
タグ機能は

``` rb
gem 'acts-as-taggable-on'
```

が便利なので調べてこちらを使って実装してみましょう。

hmiyazaki replied:
調べてみます

kuboshizuma commented on app/views/prototypes/new.html.haml:
```diff
@@ -1,37 +1,35 @@
-      %button.btn.btn-lg.btn-primary.btn-block{:type => "submit"} Publish
-      
\ No newline at end of file
+      = f.submit "送信", :class => 'btn btn-lg btn-primary btn-block'
```
最後改行いれましょうか

hmiyazaki replied:
はい。

kuboshizuma commented on db/migrate/20160108051556_create_prototypes.rb:
```diff
@@ -4,6 +4,7 @@ def change
       t.integer   :user_id
       t.string    :catchcopy
       t.text      :concept
+      t.string      :title
```
`:title` の前にインデントが入ってるのが気になります。(細かいですが）

hmiyazaki replied:
直しました。

kuboshizuma commented on test/fixtures/comments.yml:
```diff
@@ -0,0 +1,11 @@
+# Read about fixtures at http://api.rubyonrails.org/classes/ActiveRecord/FixtureSet.html
```
不要ファイルがまた入っているので消しましょう。
複数てすとファイルが入っています。

hmiyazaki replied:
自分では生成したつもりは無いのですが、どのタイミングで生成されてしまうのですか？

kuboshizuma commented:
不要ファイルやこのプルリクに不必要なファイルが多いです。
プルリクを出した後にgithub上で一度自分のファイルの変更点を見返してみると自分でミスに気づくこともあるので、一度確認する癖をつけると良いと思います。

Commit:
テスト用ファイルを消した

Commit:
commentモデルとlikeモデルを消した。
それに伴い、付随するテーブルも消した。

Commit:
インデントを揃えた

Commit:
誤って消してしまったので、再度作った。

Commit:
インデントを揃えた

Commit:
プロフィール写真が表示されなくなる問題を解決するため
・edit後の遷移先を変更してみた→効果なし。
・devise_forの行を上に移動した→効果なし

Commit:
resize_to_fitはunknownとエラーが出たので、コメントアウトした。

Commit:
最終行に、空行を追加した。
プロトタイプの画像を複数して出来るようにした。
protospaceの完成形をみたら「main_preview」というid指定のimgがあったので、追加した。

Commit:
ログイン前のユーザをリダイレクトするようにした

Commit:
不要ファイルを削除した

kuboshizuma commented:
テストファイルまだ残っています。
消しましょう。

kuboshizuma commented:
他は良さそうなので、ひとまずこのプルリクはマージして、別プルリクでtag機能をアップデートしましょう！

hmiyazaki commented:
マージしました
